src/main/java/com/gw/database/EnvironmentRepository.java

```diff
@@ -37,7 +35,7 @@ Collection<Environment> findEnvByID_BIN_ENV_BaseDir(
    * @param bin The binary path.
    * @return A collection of environment configurations matching the specified host and binary path.
    */
-  @Query(value = "select * from environment where hostid = ?1 and bin = ?2", nativeQuery = true)
+  @Query("SELECT e FROM Environment e WHERE e.hostobj.id = :hostid AND e.bin = :bin")
```

Comment:
Does this work with existing database records? I mean the records created before this PR>

Reply:
yes it works with existing h2 database